sycl/include/CL/sycl/detail/terminate_xsimk.hpp

```diff
@@ -0,0 +1,76 @@
+//==------------ sycl_mem_obj_t.hpp - SYCL standard header file ------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef __linux__
+
+inline void terminate_xsimk() {}
+
+#else
+#include <unordered_map>
+#include <unordered_set>
+#include <filesystem>
+#include <string>
+#include <fstream>
+#include <vector>
+#include <cassert>
+#include <signal.h>
+
+inline void terminate_xsimk() {
+  int my_pid = ::getpid();
+
+  /// tree of processes
```

Comment:
I have some troubles to name this :-)
```suggestion
  /// Tree of processes according to their parenthood
```